redbeams/src/main/java/com/sequenceiq/redbeams/repository/DBResourceRepository.java

```diff
@@ -25,4 +26,10 @@ public interface DBResourceRepository extends JpaRepository<DBResource, Long> {
 
     @Query("SELECT count(r) > 0 FROM DBResource r WHERE r.dbStack.id = :stackId AND r.resourceName = :name AND r.resourceType = :type")
     boolean existsByStackAndNameAndType(@Param("stackId") Long stackId, @Param("name") String name, @Param("type") ResourceType type);
+
+    @Query("SELECT r FROM DBResource r WHERE r.dbStack.id = :stackId AND r.resourceStatus = :resourceStatus AND r.resourceType = :resourceType")
+    Optional<DBResource> findFirstByResourceStatusAndResourceTypeAndDbStack(
```

Comment:
`First` seems a bit misleading as it doesn't look like from the query if there is an order by and limit